libs/x-content/src/main/java/org/elasticsearch/common/xcontent/ParsedMediaType.java

```diff
@@ -0,0 +1,139 @@
+                throw new IllegalArgumentException("invalid media type [" + headerValue + "]");
+            }
+            if (elements.length == 1) {
+                return new ParsedMediaType(splitMediaType[0].trim(), splitMediaType[1].trim(), Collections.emptyMap());
+            } else {
+                Map<String, String> parameters = new HashMap<>();
+                for (int i = 1; i < elements.length; i++) {
+                    String paramsAsString = elements[i].trim();
+                    if (paramsAsString.isEmpty()) {
+                        continue;
+                    }
+                    // intentionally allowing to have spaces around `=` (RFC disallows this)
+                    String[] keyValueParam = elements[i].trim().split("=");
+                    if (keyValueParam.length == 2) {
+                        String parameterName = keyValueParam[0].toLowerCase(Locale.ROOT).trim();
+                        String parameterValue = keyValueParam[1].toLowerCase(Locale.ROOT).trim();
+                        parameters.put(parameterName, parameterValue);
+                    } else {
+                        throw new IllegalArgumentException("invalid parameters for header [" + headerValue + "]");
+                    }
+                }
+                return new ParsedMediaType(splitMediaType[0].trim().toLowerCase(Locale.ROOT),
+                    splitMediaType[1].trim().toLowerCase(Locale.ROOT), parameters);
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Resolves this instance to a MediaType instance defined in given MediaTypeRegistry.
+     * Performs validation against parameters.
+     * @param mediaTypeRegistry a registry where a mapping between a raw media type to an instance MediaType is defined
+     * @return a MediaType instance or null if no media type could be found or if a known parameter do not passes validation
+     */
+    public  <T extends MediaType> T toMediaType(MediaTypeRegistry<T> mediaTypeRegistry) {
+        T type = mediaTypeRegistry.typeWithSubtypeToMediaType(mediaTypeWithoutParameters());
+        if (type != null) {
+
+            Map<String, Pattern> registeredParams = mediaTypeRegistry.parametersFor(mediaTypeWithoutParameters());
+            for (Map.Entry<String, String> givenParamEntry : parameters.entrySet()) {
+                if (isValidParameter(givenParamEntry.getKey(), givenParamEntry.getValue(), registeredParams) == false) {
+                    return null;
+                }
+            }
+            return type;
+        }
+        return null;
+    }
+
+    private boolean isValidParameter(String paramName, String value, Map<String, Pattern> registeredParams) {
+        if(registeredParams.containsKey(paramName)){
+            Pattern regex = registeredParams.get(paramName);
+            return regex.matcher(value).matches();
+        }
+        //TODO undefined parameters are allowed until https://github.com/elastic/elasticsearch/issues/63080
```

Comment:
can we also emit a deprecation warning ?

Reply:
how about we remove the validation in this PR?